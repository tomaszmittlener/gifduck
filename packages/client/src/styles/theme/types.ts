import { BuildContainerGrid, BuildSpanGrid } from '../grid'

export interface ThemeColorsPrimary {
  default: string
  defaultInverted: string
  secondary: string
  third: string
  [key: string]: string
}

export interface ThemeColorsAccent {
  default: string
  defaultInverted: string
  [key: string]: string
}

export interface ThemeColorsGrays {
  default: string
  white: string
  black: string
  [key: string]: string
}

export interface ThemeColorsStatus {
  success: string
  error: string
  [key: string]: string
}

export interface ThemeColors {
  primary: Readonly<ThemeColorsPrimary>
  accent: Readonly<ThemeColorsAccent>
  grays: Readonly<ThemeColorsGrays>
  status: Readonly<ThemeColorsStatus>
}

export interface ThemeTypographyFonts {
  default: string
  secondary: string
}

export interface ThemeTypographyWeights {
  regular: number
  bold: number
}
export interface ThemeTypography {
  fonts: Readonly<ThemeTypographyFonts>
  weights: Readonly<ThemeTypographyWeights>
}

export interface ThemeGrid {
  container: BuildContainerGrid
  span: BuildSpanGrid
}
export interface ThemeBreakpoints {
  tablet: string
  desktop: string
}

export interface Theme {
  colors: Readonly<ThemeColors>
  typography: Readonly<ThemeTypography>
  ms: (step: number) => string
  layers: Readonly<ThemeLayers>
  grid: Readonly<ThemeGrid>
  breakpoints: ThemeBreakpoints
}

export interface ThemeLayers {
  bottom: number
  middleBottom: number
  middle: number
  middleTop: number
  top: number
  [key: string]: number
}
