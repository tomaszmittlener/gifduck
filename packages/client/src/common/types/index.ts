export enum FetchStatus {
  Loading = 'loading',
  Loaded = 'loaded',
  Error = 'error',
}
