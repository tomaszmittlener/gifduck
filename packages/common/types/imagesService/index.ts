export interface ImageObject {
  height: string
  width: string
  url: string
}

export interface ImageData {
  preview: ImageObject
  image: ImageObject
  id: string
}

export interface ImagesResponse {
  data: ImageData[]
}

export enum SearchQueryKeys {
  searchText = 'q',
}

export interface SearchQuery {
  [SearchQueryKeys.searchText]: string
}
