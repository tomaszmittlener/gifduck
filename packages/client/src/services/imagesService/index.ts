import Axios, { AxiosPromise } from 'axios'
import { SearchQuery, ImagesSearchResponse } from '@gifduck/types/imagesService'
import { stringifyQuery } from 'common/queryString'

const BASE_URL = process.env.IMAGES_SEARCH_BAES_URL

const instance = Axios.create({
  baseURL: BASE_URL + '/api/v1/images',
})

interface ImagesService {
  getImages: (searchQuery: SearchQuery) => AxiosPromise<ImagesSearchResponse>
}

const imagesService: ImagesService = {
  getImages: async (query: SearchQuery) => {
    return instance.get(stringifyQuery(query))
  },
}

export default imagesService
