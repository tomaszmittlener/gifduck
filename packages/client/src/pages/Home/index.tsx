import React, { FunctionComponent, useEffect, useState } from 'react'
import styled from 'styled-components'
import { useLocation } from 'react-router-dom'
import { SearchQueryKeys, ImageData, SearchQuery } from '@gifduck/types/imagesService'

import { getQuery } from 'utilities/query'
import imagesService from 'services/imagesService'
import Header from 'containers/Header'
import Footer from 'containers/Footer'
import SearchResults from 'containers/SearchResults'

interface HomePageProps {
  className?: string
  testId?: string
}

const PageWrapper = styled.div`
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  background-color: ${({ theme }) => theme.colors.grays.white};
`

const Main = styled.main`
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  align-items: center;
  min-height: 50vh;
  padding: 0 0 ${({ theme }) => theme.ms(15)} 0;
`

const HomePage: FunctionComponent<HomePageProps> = ({ className, testId }) => {
  const [results, storeResults] = useState<ImageData[]>([])
  const { search } = useLocation()

  const getData = async () => {
    try {
      const searchText = getQuery(search)[SearchQueryKeys.searchText]
      const searchQuery: SearchQuery = {
        [SearchQueryKeys.searchText]: searchText ? String(searchText) : '',
      }
      const { data } = await imagesService.getImages(searchQuery)
      storeResults(data.results)
    } catch (e) {
      throw new Error(e)
    }
  }

  useEffect(() => {
    getData()
  }, [search])

  return (
    <PageWrapper data-testid={testId} className={className}>
      <Header />
      <Main>
        <SearchResults results={results} />
      </Main>
      <Footer />
    </PageWrapper>
  )
}

export default HomePage
