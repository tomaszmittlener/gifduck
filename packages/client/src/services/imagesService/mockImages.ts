import { ImagesResponse } from '@gifduck/common-types/imagesService'

const mockImages: ImagesResponse = {
  data: [
    {
      id: '1',
      image: {
        height: '500',
        url:
          'https://media2.giphy.com/media/Cmr1OMJ2FN0B2/giphy.gif?cid=e4e1ee367ecd732a2a49e8edbd6c36992b2bed00708ee5bd&rid=giphy.gif',
        width: '500',
      },
      preview: {
        width: '500',
        url:
          'https://media2.giphy.com/media/Cmr1OMJ2FN0B2/giphy_s.gif?cid=e4e1ee367ecd732a2a49e8edbd6c36992b2bed00708ee5bd&rid=giphy_s.gif',
        height: '500',
      },
    },
    {
      id: '2',
      image: {
        height: '480',
        url:
          'https://media0.giphy.com/media/65OOoKUgwJlJpRma8O/giphy.gif?cid=e4e1ee367ecd732a2a49e8edbd6c36992b2bed00708ee5bd&rid=giphy.gif',
        width: '480',
      },
      preview: {
        width: '248',
        url:
          'https://media0.giphy.com/media/65OOoKUgwJlJpRma8O/giphy-downsized_s.gif?cid=e4e1ee367ecd732a2a49e8edbd6c36992b2bed00708ee5bd&rid=giphy-downsized_s.gif',
        height: '248',
      },
    },
    {
      id: '3',
      image: {
        height: '332',
        url:
          'https://media2.giphy.com/media/vFKqnCdLPNOKc/giphy.gif?cid=e4e1ee367ecd732a2a49e8edbd6c36992b2bed00708ee5bd&rid=giphy.gif',
        width: '444',
      },
      preview: {
        width: '444',
        url:
          'https://media2.giphy.com/media/vFKqnCdLPNOKc/giphy_s.gif?cid=e4e1ee367ecd732a2a49e8edbd6c36992b2bed00708ee5bd&rid=giphy_s.gif',
        height: '332',
      },
    },
    {
      id: '4',
      image: {
        height: '353',
        url:
          'https://media0.giphy.com/media/BVStb13YiR5Qs/giphy.gif?cid=e4e1ee367ecd732a2a49e8edbd6c36992b2bed00708ee5bd&rid=giphy.gif',
        width: '500',
      },
      preview: {
        width: '500',
        url:
          'https://media0.giphy.com/media/BVStb13YiR5Qs/giphy_s.gif?cid=e4e1ee367ecd732a2a49e8edbd6c36992b2bed00708ee5bd&rid=giphy_s.gif',
        height: '353',
      },
    },
    {
      id: '5',
      image: {
        height: '358',
        url:
          'https://media3.giphy.com/media/3otPongOYeG9iINM8o/giphy.gif?cid=e4e1ee367ecd732a2a49e8edbd6c36992b2bed00708ee5bd&rid=giphy.gif',
        width: '452',
      },
      preview: {
        width: '452',
        url:
          'https://media3.giphy.com/media/3otPongOYeG9iINM8o/giphy_s.gif?cid=e4e1ee367ecd732a2a49e8edbd6c36992b2bed00708ee5bd&rid=giphy_s.gif',
        height: '358',
      },
    },
    {
      id: '6',
      image: {
        height: '260',
        url:
          'https://media2.giphy.com/media/icUEIrjnUuFCWDxFpU/giphy.gif?cid=e4e1ee367ecd732a2a49e8edbd6c36992b2bed00708ee5bd&rid=giphy.gif',
        width: '480',
      },
      preview: {
        width: '480',
        url:
          'https://media2.giphy.com/media/icUEIrjnUuFCWDxFpU/giphy_s.gif?cid=e4e1ee367ecd732a2a49e8edbd6c36992b2bed00708ee5bd&rid=giphy_s.gif',
        height: '260',
      },
    },
    {
      id: '7',
      image: {
        height: '184',
        url:
          'https://media3.giphy.com/media/IThjAlJnD9WNO/giphy.gif?cid=e4e1ee367ecd732a2a49e8edbd6c36992b2bed00708ee5bd&rid=giphy.gif',
        width: '245',
      },
      preview: {
        width: '245',
        url:
          'https://media3.giphy.com/media/IThjAlJnD9WNO/giphy_s.gif?cid=e4e1ee367ecd732a2a49e8edbd6c36992b2bed00708ee5bd&rid=giphy_s.gif',
        height: '184',
      },
    },
    {
      id: '8',
      image: {
        height: '722',
        url:
          'https://media2.giphy.com/media/YrZECW1GgBkqat6F0B/giphy.gif?cid=e4e1ee367ecd732a2a49e8edbd6c36992b2bed00708ee5bd&rid=giphy.gif',
        width: '800',
      },
      preview: {
        width: '800',
        url:
          'https://media2.giphy.com/media/YrZECW1GgBkqat6F0B/giphy_s.gif?cid=e4e1ee367ecd732a2a49e8edbd6c36992b2bed00708ee5bd&rid=giphy_s.gif',
        height: '722',
      },
    },
    {
      id: '9',
      image: {
        height: '480',
        url:
          'https://media1.giphy.com/media/MWSRkVoNaC30A/giphy.gif?cid=e4e1ee367ecd732a2a49e8edbd6c36992b2bed00708ee5bd&rid=giphy.gif',
        width: '384',
      },
      preview: {
        width: '200',
        url:
          'https://media1.giphy.com/media/MWSRkVoNaC30A/giphy-downsized_s.gif?cid=e4e1ee367ecd732a2a49e8edbd6c36992b2bed00708ee5bd&rid=giphy-downsized_s.gif',
        height: '250',
      },
    },
    {
      id: '10',
      image: {
        height: '270',
        url:
          'https://media2.giphy.com/media/3oKIPsx2VAYAgEHC12/giphy.gif?cid=e4e1ee367ecd732a2a49e8edbd6c36992b2bed00708ee5bd&rid=giphy.gif',
        width: '480',
      },
      preview: {
        width: '384',
        url:
          'https://media2.giphy.com/media/3oKIPsx2VAYAgEHC12/giphy-downsized_s.gif?cid=e4e1ee367ecd732a2a49e8edbd6c36992b2bed00708ee5bd&rid=giphy-downsized_s.gif',
        height: '216',
      },
    },
    {
      id: '11',
      image: {
        height: '360',
        url: 'https://pixabay.com/get/5ee3dd4a4253b108f5d0846096213f7d133cdbe74e507441752f7cd09f44c5_640.jpg',
        width: '640',
      },
      preview: {
        width: '150',
        url: 'https://cdn.pixabay.com/photo/2015/07/10/09/31/happy-838981_150.jpg',
        height: '84',
      },
    },
    {
      id: '12',
      image: {
        height: '640',
        url: 'https://pixabay.com/get/57e0d7444850a514f6da8c7dda793e761639dde0514c704c7d2f7bd39345cc58_640.jpg',
        width: '640',
      },
      preview: {
        width: '150',
        url: 'https://cdn.pixabay.com/photo/2015/11/06/12/43/mask-1027229_150.jpg',
        height: '150',
      },
    },
    {
      id: '13',
      image: {
        height: '360',
        url: 'https://pixabay.com/get/57e3d144425aaa14f6da8c7dda793e761639dde0514c704c7d2f7bd39345cc58_640.jpg',
        width: '640',
      },
      preview: {
        width: '150',
        url: 'https://cdn.pixabay.com/photo/2016/04/23/17/06/seal-1347886_150.jpg',
        height: '84',
      },
    },
    {
      id: '14',
      image: {
        height: '544',
        url: 'https://pixabay.com/get/57e7d2404c53a414f6da8c7dda793e761639dde0514c704c7d2f7bd39345cc58_640.png',
        width: '640',
      },
      preview: {
        width: '150',
        url: 'https://cdn.pixabay.com/photo/2016/10/27/07/53/bird-1773618_150.png',
        height: '127',
      },
    },
    {
      id: '15',
      image: {
        height: '640',
        url: 'https://pixabay.com/get/52e7d74b4e50b108f5d0846096213f7d133cdbe74e507441752f7cd09f44c5_640.jpg',
        width: '640',
      },
      preview: {
        width: '150',
        url: 'https://cdn.pixabay.com/photo/2014/10/04/11/45/cow-472842_150.jpg',
        height: '150',
      },
    },
    {
      id: '16',
      image: {
        height: '274',
        url: 'https://pixabay.com/get/54e0d4464e54aa14f6da8c7dda793e761639dde0514c704c7d2f7bd39345cc58_640.jpg',
        width: '640',
      },
      preview: {
        width: '150',
        url: 'https://cdn.pixabay.com/photo/2017/01/28/14/19/hello-2015466_150.jpg',
        height: '64',
      },
    },
    {
      id: '17',
      image: {
        height: '640',
        url: 'https://pixabay.com/get/57e2dc414355ae14f6da8c7dda793e761639dde0514c704c7d2f7bd39345cc58_640.png',
        width: '445',
      },
      preview: {
        width: '104',
        url: 'https://cdn.pixabay.com/photo/2016/03/31/15/04/alien-1292972_150.png',
        height: '150',
      },
    },
    {
      id: '18',
      image: {
        height: '360',
        url: 'https://pixabay.com/get/55e7d2444a51ab14f6da8c7dda793e761639dde0514c704c7d2f7bd39345cc58_640.jpg',
        width: '640',
      },
      preview: {
        width: '150',
        url: 'https://cdn.pixabay.com/photo/2018/10/27/17/29/window-3777037_150.jpg',
        height: '84',
      },
    },
    {
      id: '19',
      image: {
        height: '424',
        url: 'https://pixabay.com/get/5ee7d5404f50b108f5d0846096213f7d133cdbe74e507441752f7cd09f44c5_640.jpg',
        width: '640',
      },
      preview: {
        width: '150',
        url: 'https://cdn.pixabay.com/photo/2015/08/01/15/19/woman-watching-870352_150.jpg',
        height: '99',
      },
    },
    {
      id: '20',
      image: {
        height: '640',
        url: 'https://pixabay.com/get/57e8dc4a4b52a514f6da8c7dda793e761639dde0514c704c7d2f7bd39345cc58_640.jpg',
        width: '453',
      },
      preview: {
        width: '106',
        url: 'https://cdn.pixabay.com/photo/2016/12/11/10/22/businessman-1899109_150.jpg',
        height: '150',
      },
    },
  ],
}

export default mockImages
