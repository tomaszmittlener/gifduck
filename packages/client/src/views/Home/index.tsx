import React, { FunctionComponent, useEffect, useState } from 'react'
import styled from 'styled-components'
import { useLocation } from 'react-router-dom'
import { ImageData, SearchQuery, SearchQueryKeys } from '@gifduck/types/imagesService'

import { getQuery } from 'common/queryString'
import imagesService from 'services/imagesService'
import Header from 'containers/Header'
import Footer from 'containers/Footer'
import SearchResults from 'containers/SearchResults'
import { FetchStatus } from 'common/types'
import scrollToTop from 'common/scrollToTop'

interface HomePageProps {
  className?: string
  testId?: string
}

const PageWrapper = styled.div`
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  background-color: ${({ theme }) => theme.colors.grays.white};
`

const Main = styled.main`
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  align-items: center;
  min-height: 50vh;
  padding: 0 0 ${({ theme }) => theme.ms(15)} 0;
`

const HomePage: FunctionComponent<HomePageProps> = ({ className, testId }) => {
  const [results, storeResults] = useState<ImageData[]>([])
  const [fetchStatus, updateFetchStatus] = useState<FetchStatus>(FetchStatus.Loading)

  const { search } = useLocation()

  const getData = async () => {
    try {
      updateFetchStatus(FetchStatus.Loading)

      const searchText = getQuery(search)[SearchQueryKeys.searchText]
      const searchQuery: SearchQuery = {
        [SearchQueryKeys.searchText]: searchText ? String(searchText) : '',
      }
      const { data } = await imagesService.getImages(searchQuery)
      storeResults(data.results)
      updateFetchStatus(FetchStatus.Loaded)

      scrollToTop()
    } catch (e) {
      updateFetchStatus(FetchStatus.Error)

      throw new Error(e)
    }
  }

  useEffect(() => {
    getData()
  }, [search])

  return (
    <PageWrapper data-testid={testId} className={className}>
      <Header isLoading={fetchStatus === FetchStatus.Loading} />
      <Main>
        <SearchResults results={results} />
      </Main>
      <Footer />
    </PageWrapper>
  )
}

export default HomePage
