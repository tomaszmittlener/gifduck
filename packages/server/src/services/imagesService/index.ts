import { ImageData, SearchQuery } from '@gifduck/types/imagesService'
import giphyService from './utilities/giphyService'
import pixabayService from './utilities/pixabayService'
import formatGiphyResponse from './utilities/formatGiphyResponse'
import formatPixabayResponse from './utilities/formatPixabayResponse'

const imagesService = {
  search: async (searchQuery: SearchQuery): Promise<ImageData[]> => {
    try {
      const giphyData = await giphyService.search(searchQuery)
      const giphyImages = formatGiphyResponse(giphyData.data)
      const pixabayData = await pixabayService.search(searchQuery)
      const pixabayImages = await formatPixabayResponse(pixabayData.data)

      return [...giphyImages, ...pixabayImages]
    } catch (e) {
      throw e
    }
  },
}

export default imagesService
