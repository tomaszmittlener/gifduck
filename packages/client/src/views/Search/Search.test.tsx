import React from 'react'
import { render, cleanup, wait } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import Theme from 'styles/theme'

import Home from './'
import { MemoryRouter } from 'react-router-dom'
import { ImagesSearchResponse } from '@gifduck/types/imagesService'

const defaultProps = {
  className: 'class-name',
  testId: 'test-id',
}

afterEach(cleanup)

const response: ImagesSearchResponse = {
  results: [
    {
      preview: { aspectRatio: 1, url: 'prev-url', height: 10, width: 10 },
      image: { width: 20, height: 20, url: 'img-url', aspectRatio: 1 },
      id: '1',
    },
  ],
}

// mock search results response
jest.mock('services/imagesService', () => ({
  esModule: true,
  getImages: async () => ({ data: response }),
}))

// mock window scrollTo fn
window.scrollTo = jest.fn()

describe('Views - Search', () => {
  test('renders correctly', async () => {
    const renderComponent = render(
      <ThemeProvider theme={Theme}>
        <MemoryRouter>
          <Home />
        </MemoryRouter>
      </ThemeProvider>,
    )

    await wait(() => renderComponent)

    expect(renderComponent.container).toMatchSnapshot()
  })

  test('passes className to container', async () => {
    const renderComponent = render(
      <ThemeProvider theme={Theme}>
        <MemoryRouter>
          <Home {...defaultProps} />
        </MemoryRouter>
      </ThemeProvider>,
    )

    await wait(() => renderComponent)

    expect(renderComponent.getByTestId(defaultProps.testId)).toHaveClass(defaultProps.className)
  })
})
