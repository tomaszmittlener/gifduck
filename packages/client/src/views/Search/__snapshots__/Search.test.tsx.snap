// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Views - Search renders correctly 1`] = `
.c3 {
  font-size: 2.2806973457336426rem;
  font-family: Pacifico,sans-serif;
  color: #F2C94C;
  line-height: 2;
  width: auto;
  -webkit-animation: dGzoaH 1s ease-out none;
  animation: dGzoaH 1s ease-out none;
}

.c1 {
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  width: 100%;
  z-index: 1500;
  background-color: #343339;
}

.c2 {
  padding: 0 1rem;
  grid-column-gap: 0.49327018427257213rem;
  display: grid;
  grid-template-columns: repeat(8,1fr);
  width: 100%;
  max-width: 1440px;
  margin: 0 auto;
}

.c5 {
  grid-column: 1 / 9;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 0 1.601806640625rem 0;
}

.c7 {
  height: 3.247321025468409rem;
  width: 3.247321025468409rem;
  border: none;
  background-color: #ECE9DE;
  padding: 0;
  margin: 0;
  cursor: pointer;
  border-radius: 0 10px 10px 0;
}

.c6 {
  border: none;
  line-height: 1.75;
  height: 3.247321025468409rem;
  color: #5D899E;
  background-color: #ECE9DE;
  font-size: 1rem;
  font-family: Open Sans,sans-serif;
  padding: 0 1rem;
  width: 100%;
  border-radius: 10px 0 0 10px;
}

.c4 {
  padding: 0.7901234567901234rem 0;
  grid-column: 1 / 9;
  justify-self: center;
}

.c15 {
  margin: auto 0 0 0;
  width: 100%;
  min-height: 19.00260151070504rem;
  background-color: #F2C94C;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 1.423828125rem 0px;
}

.c16 {
  color: #706155;
  font-family: Pacifico,sans-serif;
  font-size: 0.7901234567901234rem;
}

.c10 {
  padding: 0;
  display: grid;
  grid-row-gap: 0.8888888888888888rem;
  grid-column-gap: 1%;
  grid-template-columns: 1fr;
}

.c12 {
  padding: 0;
  margin: 0;
  list-style: none;
  grid-area: auto / 1 / span 2 / 5;
}

.c13 {
  margin: 0;
  position: relative;
  width: 100%;
  height: 100%;
}

.c13:before {
  display: block;
  content: '';
  width: 100%;
  padding-top: 100%;
}

.c14 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 1;
  z-index: 1100;
}

.c9 {
  padding: 0 1rem;
  grid-column-gap: 0.49327018427257213rem;
  display: grid;
  grid-template-columns: repeat(8,1fr);
  width: 100%;
  max-width: 1440px;
  height: 100%;
}

.c11 {
  grid-column: 1 / 9;
}

.c0 {
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #ECE9DE;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 50vh;
  padding: 0 0 5.851777930691043rem 0;
}

@media screen and (min-width:768px) {
  .c2 {
    padding: 0 1.423828125rem;
  }
}

@media screen and (min-width:1024px) {
  .c2 {
    padding: 0 2.0272865295410156rem;
  }
}

@media screen and (min-width:768px) {
  .c2 {
    grid-column-gap: 1rem;
  }
}

@media screen and (min-width:768px) {
  .c4 {
    justify-self: left;
    grid-column: 1 / 4;
  }
}

@media screen and (min-width:768px) {
  .c10 {
    grid-row-gap: 1.601806640625rem;
    grid-template-columns: 6% 14% 11% 15% 5% 10% 9% 23%;
  }
}

@media screen and (min-width:768px) {
  .c12:first-of-type {
    grid-area: auto / 3 / span 1 / 7;
  }

  .c12:nth-of-type(2n) {
    grid-area: auto / 4 / span 1 / 9;
  }

  .c12:nth-of-type(3n) {
    grid-area: auto / 2 / span 1 / 7;
  }

  .c12:nth-of-type(4n) {
    grid-area: auto / 5 / span 1 / 9;
  }

  .c12:nth-of-type(7n) {
    grid-area: auto / 4 / span 1 / 6;
  }
}

@media screen and (min-width:768px) {
  .c9 {
    padding: 0 1.423828125rem;
  }
}

@media screen and (min-width:1024px) {
  .c9 {
    padding: 0 2.0272865295410156rem;
  }
}

@media screen and (min-width:768px) {
  .c9 {
    grid-column-gap: 1rem;
  }
}

<div>
  <div
    class="c0"
  >
    <header
      class="c1"
    >
      <div
        class="c2"
      >
        <label
          class="c3 c4"
        >
          üê§gifduck
        </label>
        <form
          class="c5"
        >
          <input
            class="c6"
            name="search"
            placeholder="Search for Images..."
            type="text"
          />
          <button
            class="c7"
          >
            üîé
          </button>
        </form>
      </div>
    </header>
    <main
      class="c8"
    >
      <section
        class="c9"
      >
        <ul
          class="c10 c11"
        >
          <li
            class="c12"
          >
            <figure
              class="c13"
            >
              <img
                class="c14"
                src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkqAcAAIUAgUW0RjgAAAAASUVORK5CYII="
              />
            </figure>
          </li>
        </ul>
      </section>
    </main>
    <footer
      class="c15"
    >
      <span
        class="c16"
      >
        @gifduck
      </span>
    </footer>
  </div>
</div>
`;
